ssllib.c

```diff
@@ -258,27 +258,33 @@ ssl_inspect_status(PL_SSL_INSTANCE *instance, int ssl_ret, status_role role)
       break;
   }
 
+  if ( ( code == SSL_ERROR_SYSCALL ) &&
+       ( BIO_eof(SSL_get_rbio(instance->ssl)) ||
+         BIO_eof(SSL_get_wbio(instance->ssl))) )
+  { switch(role)
+    { case STAT_NEGOTIATE:
+        PL_raise_exception(unexpected_eof(instance));
```

Comment:
This is not correct.  If unexpected_eof() cannot create the exception this will crash.   The way to handle this is roughly like below.  Now, if
there is no space for the exception you'll nicely end up with the appropriate resource error.  Of course you can generalise this away.

```
term_t ex;

return ( ex = PL_new_term_ref() &&
         PL_unify_term(ex, ....) &&
         PL_raise_exception(ex) );
```